blockchain/vm/contracts_test.go

```diff
@@ -56,6 +56,25 @@ type precompiledFailureTest struct {
 	Name          string
 }
 
+// allPrecompiles does not map to the actual set of precompiles, as it also contains
+// repriced versions of precompiles at certain slots
+var allPrecompiles = map[common.Address]PrecompiledContract{
+	common.BytesToAddress([]byte{1}):    &ecrecover{},
+	common.BytesToAddress([]byte{2}):    &sha256hash{},
+	common.BytesToAddress([]byte{3}):    &ripemd160hash{},
+	common.BytesToAddress([]byte{4}):    &dataCopy{},
+	common.BytesToAddress([]byte{5}):    &bigModExp{eip2565: false},
+	common.BytesToAddress([]byte{0xf5}): &bigModExp{eip2565: true},
```

Comment:
If this line is deleted, this allPrecompiles is same as PrecompiledContractsIstanbulCompatible. Could you explain why you insert a line 67 in this test because I don't find the use case this line?

Reply:
As long as i know, it is for testing the modexp precompiled contract. 
* if eip2565 is false, it tests the precompiled contract which is used before eip2565.
* if eip2565 is true, it tests the precompiled contract which is used after eip25656.

You can see here.
```
// Tests the sample inputs from the ModExp EIP 198.
func TestPrecompiledModExp(t *testing.T)      { testJson("modexp", "05", t) }
func BenchmarkPrecompiledModExp(b *testing.B) { benchJson("modexp", "05", b) }

func TestPrecompiledModExpEip2565(t *testing.T)      { testJson("modexp_eip2565", "f5", t) }
func BenchmarkPrecompiledModExpEip2565(b *testing.B) { benchJson("modexp_eip2565", "f5", b) }
```